napari/layers/shapes/_shapes_models/shape.py

```diff
@@ -413,14 +425,18 @@ def to_mask(self, mask_shape=None, zoom_factor=1, offset=(0, 0)):
         # and embed as a slice.
         if embedded:
             mask = np.zeros(mask_shape, dtype=bool)
-            slice_key = [0] * len(mask_shape)
+            slice_key: list[int | slice] = [0] * len(mask_shape)
             for i in range(len(mask_shape)):
                 if i in self.dims_displayed:
                     slice_key[i] = slice(None)
-                else:
+                elif self.slice_key is not None:
                     slice_key[i] = slice(
                         self.slice_key[0, i], self.slice_key[1, i] + 1
                     )
+                else:
+                    raise RuntimeError(
+                        "Internal error: self.slice_key is None"
+                    )
```

Comment:
Since the above clause would error anyway if `self.slice_key == None`, I figured it was fine to put in an error thrown here to make the static typing work.